### DISCUSSION TITLE
[New Feature] "Seek transports" hotkey

### DISCUSSION THREAD

**Discussion by CCTiberius** [Vinifera-Developers/Vinifera#1414] [Feature Requests]
### Description:
A hotkey that makes units seek out and enter transports within their guard range. The point is to make is much more convenient to load several transports with a single hotkey press, instead of the rather clunky current method of filling them one by one with substantial waiting time.

### Possible Implementation:
A hotkey and the function would just tie into the unit's guard range so no additional INI keys are needed

**Comment by CCHyper** [comment]
I foresee three issues;
1) How should the game act when there is an APC _(Unit)_ and an Air Transport _(Aircraft)_ within its range?
2) How should the game act when there are two means of transport of the same type within its range?
3) How should the game act when there are two means of transport at the same distance within its range?

**Comment by Rampastring** [comment]
For 1 and 2 I think it'd be most intuitive for each unit to prefer the transport that is closest. Unless it's full.

For 3, it'd probably be easiest to send to either via RNG and it'd work well enough in practice. Another option is allowing defining the priority of transports somehow, or prioritize them automatically based on RTTI and object type index (while not as easily customizable, this would at least make the behaviour always consistent for the player).

**Comment by Onidotmoe** [comment]
Instead of a dedicated hotkey, this could be implemented by Shift-Double-Clicking(or Double-Right-Clicking? or Alt-Clicking?) the transport with the units selected and the units will try to enter all the transports of that type that they can get to in the current screen.

Or with Alt-D(like the Deploy key but reversed) forcing units to go into the nearest transport, relative to themselves inside the current screen.
Players would probably be inclined to only have the units near transport that they specifically want to get them into but there could be bias added to it.

Bias Hierarchy : 
Already Has Units inside and Has Room for the Unit ordered to enter that transport > Most Health > Ground Transport > Subterranean Transport > Air Transport

Tho not a perfect representation but if we say that first there has to be room for the unit, then it checks if there are already units inside, if so, chose that over another transport that's also next to it but is empty.
If both criteria are the same the transport with the most health will be chosen.
If it's still the same then the ground transport will be chosen first if the other options are anything else.
If all transports are of the same type, then the north-most transport will be picked first, and from there it would go clockwise around till it reaches north-west and loops back to north.
Could also add a bias for if there are units of the same type inside the transport, so engineers would want to enter vehicles that already has engineers in it, or try to group up and enter the same vehicle together.
Or units with no weapons would enter a vehicle together.

I feel like adding units to on-surface transports like apcs is most common and also what you'd do when you want to move them along with other units, where Subterranean and Air Transports are more like something you'd need to focus and pay more attention to when you move them as they're usually not as protected.
